skbio/alignment/_tabular_msa.py

```diff
@@ -1701,7 +1701,15 @@ def reassign_index(self, mapping=None, minter=None):
                 "Cannot use both `mapping` and `minter` at the same time.")
 
         if mapping is not None:
-            self._seqs.rename(mapping, inplace=True)
+            if isinstance(mapping, dict):
+                self.index = [mapping[label] if label in mapping else label
```

Comment:
Developer preference I think. My preference probably matches the ternary as it looks a little "prettier" and it's purpose feels clearer in this context. But there's nothing wrong with your approach either.